toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -493,6 +495,22 @@ func PopulateInstallRoot(installChroot *safechroot.Chroot, packagesToInstall []s
 		return
 	}
 
+	// imageconfigvalidator should have ensured that we intend to install shadow-utils, so we can go ahead and do that here.
+	if len(config.Users) > 0 || len(config.Groups) > 0 {
+		if !sliceutils.ContainsValue(packagesToInstall, "shadow-utils") {
+			err = fmt.Errorf("shadow-utils package is required when setting users or groups")
+			return
+		}
+
+		packagesInstalled, err = TdnfInstallWithProgress(shadowUtilsPkg, installRoot, packagesInstalled, totalPackages, true)
+		if err != nil {
+			return
+		}
+
+		// Remove shadow-utils from the list of packages to install
+		packagesToInstall = sliceutils.FindMatches(packagesToInstall, func(pkg string) bool { return pkg != shadowUtilsPkg })
```

Comment:
Why do we need to install `shadow-utils` separately? And why do we remove it later from `packagesToInstall`?

Reply:
The point of this change is to add users before installing any other packages. To do that we need shadow-utils (so at best we can try to minimize the number of other packages added). It feels cleaner to not re-install shadowutils if its already installed.